### PR TITLE
update toast documentation on accessibility

### PR DESCRIPTION
**Pull request by Lausselloic**
update documentation following discussion with @patrickhlauke  on PR #27155 

Maybe need a review for english because it's not my native langage.

Also add a question about default toast timeout actually defined to `500ms` maybe most of users will let it as is by copy/paste... don't know what's the usual timer on toaster but maybe increase it by default at 3seconds or 5seconds?

### CODE REVIEW

**Comment by Johann-S** [comment]
About the timeout I chose that value `500ms` because for small message it'll be enough and for long message it won't be enough.
So we have to help our users to think by themselves we are not their mother/father and they are developers.
You added a useful information about to choose the right timeout so that's enough 👍

Personnaly the only doubts I have, it's, is it useful to have a show delay ?

File: site/docs/4.1/components/toasts.md [hunk 1]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
-To improve accessibility level, we strongly recomend to use `autohide: false` and add a `close` button into the header to let user dismiss that element.
-You also need to adapt the `role` and `aria-live` level depending on the content. If it's an important message like error, use an `alert` role `assertive` otherwise use a role `status` with a `polite` level. 
+Toasts are intended to be small interruptions to your visitors or users, so to help those on screen readers, you should wrap your toasts in an [`aria-live` region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). This allows screen readers the ability to see suggested interruptions without any visual cues. If the information need is important for the process, like list of form fields in error, then use the [alert component]({{ site.baseurl }}/docs/{{ site.docs_version }}/components/alerts/) instead of toast.
```
Comment (notabene) [inline site/docs/4.1/components/toasts.md]: s/need/needed/
s/like list of form fields in error/e.g. for a list of for errors in a form/

**Review by notabene — CHANGES_REQUESTED** [review]

File: site/docs/4.1/components/toasts.md [hunk 2]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
+
+You also need to adapt the `role` and `aria-live` level depending on the content. If it's an important message like error, use `role="alert" aria-live="assertive"`, otherwise use `role="status" aria-live="polite"` attributes.
```
Comment (notabene) [inline site/docs/4.1/components/toasts.md]: s/an important message like error/an important message like an error/

File: site/docs/4.1/components/toasts.md [hunk 3]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
+
+Be carefull to adapt the [`hide` timeout](#options) depending on the content to let all user enough time to read the content.
```
Comment (notabene) [inline site/docs/4.1/components/toasts.md]: s/let all user/give all users/

**Review by notabene — CHANGES_REQUESTED** [review]

File: site/docs/4.1/components/toasts.md [hunk 4]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
 
 {% highlight html %}
-<div role="alert" aria-live="assertive" aria-atomic="true">
+<div role="alert" aria-live="assertive" aria-atomic="true" data-delay='{"show": 0, "hide": 10000}'>
   <div role="alert" aria-live="assertive" aria-atomic="true">...</div>
 </div>
 {% endhighlight %}
 
+If you use `autohide: false` then you need to add a `close` button into the header to let user dismiss that element.
```
Comment (notabene) [inline site/docs/4.1/components/toasts.md]: s/to let user dismiss/to enable users to dismiss/

**Review by notabene — CHANGES_REQUESTED** [review]

**Comment by XhmikosR** [comment]
@mdo @patrickhlauke: LGTY?

**Comment by Johann-S** [comment]
>Personnaly the only doubts I have, it's, is it useful to have a show delay ?

What do you think about that @XhmikosR ?

**Comment by XhmikosR** [comment]
@Johann-S: not sure to be honest, I'm not sure the branch is stable enough yet.

BTW I rebased this and the base branch to resolve the conflicts.

**Comment by Johann-S** [comment]
@Lausselloic I updated the main PR by removing the show delay and fixing our documentation, can you rebase your branch ?

**Comment by Johann-S** [comment]
finally I did it 😆

File: site/docs/4.1/components/toasts.md [hunk 5]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
-You also need to adapt the `role` and `aria-live` level depending on the content. If it's an important message like error, use an `alert` role `assertive` otherwise use a role `status` with a `polite` level. 
+Toasts are intended to be small interruptions to your visitors or users, so to help those on screen readers, you should wrap your toasts in an [`aria-live` region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions). This allows screen readers the ability to see suggested interruptions without any visual cues. If the information needed is important for the process, e.g. for a list of errors in a form, then use the [alert component]({{ site.baseurl }}/docs/{{ site.docs_version }}/components/alerts/) instead of toast.
+
+You also need to adapt the `role` and `aria-live` level depending on the content. If it's an important message like an error, use `role="alert" aria-live="assertive"`, otherwise use `role="status" aria-live="polite"` attributes.
+
+Be carefull to adapt the [`delay` timeout](#options) depending on the content to give all users enough time to read the content.
```
Comment (mdo) [inline site/docs/4.1/components/toasts.md]: Noticed a typo here, but decided to rewrite this bit. Can you change this? If not, I can push a commit.

```
As the content you're displaying changes, be sure to update the [`delay` timeout](#options) to ensure people have enough time to read the toast.
```

File: site/docs/4.1/components/toasts.md [hunk 6]
```diff
@@ -99,16 +93,39 @@ Plus, they'll easily stack.
...
 
 {% highlight html %}
-<div role="alert" aria-live="assertive" aria-atomic="true">
+<div class="toast" role="alert" aria-live="assertive" aria-atomic="true" data-delay="10000">
   <div role="alert" aria-live="assertive" aria-atomic="true">...</div>
 </div>
 {% endhighlight %}
 
+If you use `autohide: false` then you need to add a `close` button into the header to enable users to dismiss that element.
```
Comment (mdo) [inline site/docs/4.1/components/toasts.md]: Change to:

```
When using `autohide: false`, you must add a close button to allow users to dismiss the toast.
```

**Review by mdo — DISMISSED** [review]
Couple text changes and then this seems good to go.

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**